src/geocat/f2py/linint2_wrapper.py

```diff
@@ -296,61 +282,61 @@ def linint2(fi, xo, yo, xi=None, yi=None, icycx=0, msg_py=None):
         will have the same dimension sizes as the lengths of yo and xo.
         The return type will be double if fi is double, and float
         otherwise.
-
     Examples
     --------
-
     Example 1: Using linint2 with :class:`xarray.DataArray` input
-
     .. code-block:: python
-
         import numpy as np
         import xarray as xr
         import geocat.comp
-
         fi_np = np.random.rand(30, 80)  # random 30x80 array
-
         # xi and yi do not have to be equally spaced, but they are
         # in this example
         xi = np.arange(80)
         yi = np.arange(30)
-
         # create target coordinate arrays, in this case use the same
         # min/max values as xi and yi, but with different spacing
         xo = np.linspace(xi.min(), xi.max(), 100)
         yo = np.linspace(yi.min(), yi.max(), 50)
-
         # create :class:`xarray.DataArray` and chunk it using the
         # full shape of the original array.
         # note that xi and yi are attached as coordinate arrays
         fi = xr.DataArray(fi_np,
                           dims=['lat', 'lon'],
                           coords={'lat': yi, 'lon': xi}
                          ).chunk(fi_np.shape)
-
         fo = geocat.comp.linint2(fi, xo, yo, icycx=0)
-
     """
 
     # ''' Start of boilerplate
-    if not isinstance(fi, xr.DataArray):
-        if (xi is None) | (yi is None):
-            raise CoordinateError(
-                "linint2: Arguments xi and yi must be provided explicitly unless fi is an xarray.DataArray."
-            )
+    # Check data type for input variables
+    check(xo, data_type=np.ndarray) # ToDo: check necessity
+    check(yo, data_type=np.ndarray)
+
+    if (xi is not None) | (yi is not None):
```

Comment:
If `fi` is `Xarray` already (which is checked for in Line 320), we are not really concerned with `xi` and `yi`. Our concern for `xi` and `yi` to be `np.ndarray` starts when `fi` is not xarray. Therefore, these check for `xi` and `yi` are redundant here. It is enough to check them only after we see `fi` is not xarray, i.e. it is numpy. This is possible starting Line 322.